lib/pip.js

```diff
@@ -194,11 +282,129 @@ function copyVendors(vendorFolder, targetFolder, serverless) {
+
+  installRequirementsFile(
+    fileName,
+    slsReqsTxt,
+    serverless,
+    servicePath,
+    options
+  );
+
+  // If no requirements file or an empty requirements file, then do nothing
+  if (!fse.existsSync(slsReqsTxt) || fse.statSync(slsReqsTxt).size == 0) {
+    serverless.cli.log(
+      `Skipping empty output requirements.txt file from ${slsReqsTxt}`
+    );
+    return false;
+  }
+
+  // Copy our requirements to another filename in .serverless (incase of individually packaged)
+  if (modulePath && modulePath != '.') {
+    fse.existsSync(path.join(servicePath, '.serverless', modulePath));
+    const destinationFile = path.join(
+      servicePath,
+      '.serverless',
+      modulePath,
+      'requirements.txt'
+    );
+    serverless.cli.log(
+      `Copying from ${slsReqsTxt} into ${destinationFile} ...`
+    );
+    fse.copySync(slsReqsTxt, destinationFile);
+  }
+
+  // Then generate our MD5 Sum of this requirements file to determine where it should "go" to and/or pull cache from
+  const reqChecksum = md5Path(slsReqsTxt);
+
+  // Then figure out where this cache should be, if we're caching, if we're in a module, etc
+  const workingReqsFolder = getRequirementsWorkingPath(
+    reqChecksum,
+    servicePath,
+    options
+  );
+
+  // Check if our static cache is present and is valid
+  if (fse.existsSync(workingReqsFolder)) {
+    if (
+      fse.existsSync(path.join(workingReqsFolder, '.completed_requirements')) &&
+      workingReqsFolder.endsWith('_slspyc')
+    ) {
+      serverless.cli.log(
+        `Using static cache of requirements found at ${workingReqsFolder} ...`
+      );
+      // We'll "touch" the folder, as to bring it to the start of the FIFO cache
+      fse.utimesSync(workingReqsFolder, new Date(), new Date());
+      return workingReqsFolder;
+    }
+    // Remove our old folder if it didn't complete properly, but _just incase_ only remove it if named properly...
+    if (
+      workingReqsFolder.endsWith('_slspyc') ||
+      workingReqsFolder.endsWith('.requirements')
+    ) {
+      rimraf.sync(workingReqsFolder);
+    }
+  }
+
+  // Ensuring the working reqs folder exists
+  fse.ensureDirSync(workingReqsFolder);
+
+  // Copy our requirements.txt into our working folder...
+  fse.copySync(slsReqsTxt, path.join(workingReqsFolder, 'requirements.txt'));
+
+  // Then install our requirements from this folder
+  installRequirements(workingReqsFolder, serverless, options);
+
+  // Copy vendor libraries to requirements folder
+  if (options.vendor) {
+    copyVendors(options.vendor, workingReqsFolder, serverless);
+  }
+  if (funcOptions.vendor) {
+    copyVendors(funcOptions.vendor, workingReqsFolder, serverless);
+  }
+
+  // Then touch our ".completed_requirements" file so we know we can use this for static cache
+  if (options.useStaticCache) {
+    fse.closeSync(
+      fse.openSync(path.join(workingReqsFolder, '.completed_requirements'), 'w')
+    );
+  }
```

Comment:
@AndrewFarley I think I know what's happening here. Your code is actually working for the `useDownloadCache`, I failed to notice it because the package I'm installing requires a lot of compiling, but I logged the docker container logs and this part is actually working. 👍 

What fails to work is `useStaticCache`, something is removing the folder that `.endWith('_slspyc')`, along with the `.completed_requirements` file that you use to verify that the static cache actually exists.

I tested this by adding `throw new Error()` here. The code fails but the folder remains!!!

![image](https://user-images.githubusercontent.com/9551950/41384733-e9d80438-6f4d-11e8-98a3-b06bf1944109.png)

Then I removed the error and tried again, and the cached worked! But then something erases the folder.

```sh
Serverless: Using static cache of requirements found at /Users/felipesantos/apps/sls-pip-cache/cached_requirements/ee028457a1ea41250e8af2a379765ce5_slspyc ...
Serverless: Zipping required Python packages for func1...
Serverless: Packaging service...
Serverless: Excluding development dependencies...
```

![image](https://user-images.githubusercontent.com/9551950/41384797-43479e02-6f4e-11e8-8d79-a658239f50f7.png)

I think serverless is erasing it, but idk how to prevent. Do you know?

Thanks!